test/parallel/test-stringbytes-external-at-max.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+
+// v8 fails silently if string length > v8::String::kMaxLength
+(function() {
+  // v8::String::kMaxLength defined in v8.h
+  const kStringMaxLength = process.binding('buffer').kStringMaxLength;
+
+  try {
+    new Buffer(kStringMaxLength * 3);
+  } catch(e) {
+    assert.equal(e.message, 'Invalid array buffer length');
+    console.log(
+        '1..0 # Skipped: intensive toString tests due to memory confinements');
+    return;
+  }
+
+  const buf = new Buffer(kStringMaxLength);
+
+  var maxString = buf.toString();
+  assert.equal(maxString.length, kStringMaxLength);
```

Comment:
Not a good test. The memory needs to be filled or else it may contain random multi-byte characters and the string length might vary. Example:
```js
let b = new Buffer(4).fill('a');
b[0] = 0xc8;
b[1] = 0xa2;
b.toString().length === 3;
```
So as much as I hate to do this, fill the above buffer with `0x61` (numbers are currently optimized better than strings). This way you get reliable results.